### PR TITLE
Remove checkbox label from inside code block

### PR DESCRIPTION
**Pull request by AlanGabbianelli**
Having the `label` tag inside a code block within the `input` tag makes
the generated html have the `label` element wrapped inside an `input`
element.

This occurs when running `bundle exec rake default` in `finder-frontend`
(which is the command the Jenkins runs). Running `bundle exec cucumber`
does not present the same issue.

The problematic `input` element has both an opening tag and a closing
tag, causing parsing errors in strict mode in this `slimmer` method:
```
def parse_html(html, description_for_error_message)
  doc = Nokogiri::HTML.parse(html)
  if strict
    errors = doc.errors.select {|e| e.error?}.reject {|e| ignorable?(e)}
    if errors.size > 0
      error = errors.first
      message = "In #{description_for_error_message}: '#{error.message}' 
at line #{error.line} col #{error.column} (code #{error.code}).\n"
      message << "Add ?skip_slimmer=1 to the url to show the raw backend 
request.\n\n"
      message << context(html, error)
      raise message
    end
  end

  doc
end
```

The error returned is:
```
In backend response: '66:9: ERROR: Unexpected end tag : input' at line 
66 col 9 (code 76).
Add ?skip_slimmer=1 to the url to show the raw backend request.

  61:     <div class="govuk-checkboxes">
  62:
  63: <div class="gem-c-checkbox govuk-checkboxes__item">
  64:   <input id="checkboxes-6088ba3d-0" name="aircraft_category[]" 
type="checkbox" value="commercial-fixed-wing" 
class="govuk-checkboxes__input">
  65:     <label class="govuk-label govuk-checkboxes__label" 
for="checkboxes-6088ba3d-0">Commercial - fixed wing</label>
        -----v
  66: </input></div>
  67:
```

Co-Authored-By: edwardkerry <edward.kerry@digital.cabinet-office.gov.uk>

Remember to update the CHANGELOG if your change needs to be listed in the next version of the gem.

---

Component guide for this PR:
https://govuk-publishing-compon-pr-[THIS PR NUMBER].herokuapp.com/component-guide/

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -332,5 +332,8 @@ DEPENDENCIES
   webmock (~> 3.0.1)
   yard
 
+RUBY VERSION
```
Comment (barrucadu) [inline Gemfile.lock]: It would be good to not have this unrelated change

Comment (AlanGabbianelli) [reply Gemfile.lock]: We left it because it was a small change with little impact, but we can remove it if you think it's necessary.
Were you thinking of putting it in a different commit or removing it completely from this PR @barrucadu ?

Comment (barrucadu) [reply Gemfile.lock]: I think just remove it entirely

Comment (AlanGabbianelli) [reply Gemfile.lock]: Can do

**Comment by steventux** [comment]
Is it the case that `html` is malformed before it is passed to `Nokogiri::HTML.parse`? ie. that the ERB templates render malformed HTML? Or is it Nokogiri's interpretation that fails to validate this string?
I ask because we probably shouldn't change the implementation of a component based on a limitation in Nokogiri. 
If ERB is generating invalid markup because of the tag nesting here then I think this is a sensible change.
